python/hail/methods/family_methods.py

```diff
@@ -188,24 +185,109 @@ def mendel_errors(dataset, pedigree) -> Tuple[Table, Table, Table, Table]:
+
     Parameters
     ----------
     dataset : :class:`.MatrixTable`
-        Dataset.
     pedigree : :class:`.Pedigree`
-        Sample pedigree.
 
     Returns
     -------
     (:class:`.Table`, :class:`.Table`, :class:`.Table`, :class:`.Table`)
-        Four tables as detailed in notes with Mendel error statistics.
     """
-
-    dataset = require_biallelic(dataset, 'mendel_errors')
-
-    kts = dataset._jvds.mendelErrors(pedigree._jrep)
-    return Table(kts._1()), Table(kts._2()), \
-           Table(kts._3()), Table(kts._4())
+    source = call._indices.source
+    if not isinstance(source, MatrixTable):
+        raise ValueError("'mendel_errors': expected 'call' to be an expression of 'MatrixTable', found {}".format(
+            "expression of '{}'".format(source.__class__) if source is not None else 'scalar expression'))
+
+    source = source.select_entries(__GT=call)
+    dataset = require_biallelic(source, 'mendel_errors')
+    tm = trio_matrix(dataset, pedigree, complete_trios=True)
+    tm = tm.select_entries(mendel_code=hl.mendel_error_code(
+        tm.locus,
+        tm.is_female,
+        tm.father_entry['__GT'],
+        tm.mother_entry['__GT'],
+        tm.proband_entry['__GT']
+    ))
+    tm = tm.filter_entries(hl.is_defined(tm.mendel_code))
+
+    ck_name = next(iter(source.col_key))
+
+    entries = tm.entries()
+
+    table1 = entries.select(*tm.row_key, *tm.col_key, 'fam_id', 'mendel_code')
+
+    fam_counts = (
+        entries
+            .group_by(pat_id=entries.father[ck_name], mat_id=entries.mother[ck_name])
+            .partition_hint(min(entries.n_partitions(), 8))
+            .aggregate(errors=hl.agg.count_where(hl.is_defined(entries.mendel_code)),
+                       snp_errors=hl.agg.count_where(hl.is_snp(entries.alleles[0], entries.alleles[1]) &
+                                                     hl.is_defined(entries.mendel_code)),
+                       children=hl.len(hl.agg.collect_as_set(entries.id)))
+    )
+    table2 = tm.cols()
+    table2 = table2.select(pat_id=table2.father[ck_name],
+                           mat_id=table2.mother[ck_name],
+                           fam_id=table2.fam_id,
+                           **fam_counts[table2.father[ck_name], table2.mother[ck_name]])
+    table2 = table2.key_by('mat_id', 'pat_id').distinct()
+    table2 = table2.annotate(errors=hl.or_else(table2.errors, hl.int64(0)),
+                             snp_errors=hl.or_else(table2.snp_errors, hl.int64(0)))
+
+    # in implicated, idx 0 is dad, idx 1 is mom, idx 2 is child
+    implicated = hl.array([
+        [0, 0, 0],  # dummy
+        [1, 1, 1],
+        [1, 1, 1],
+        [1, 0, 1],
+        [0, 1, 1],
+        [0, 0, 1],
+        [1, 0, 1],
+        [0, 1, 1],
+        [0, 0, 1],
+        [0, 1, 1],
+        [0, 1, 1],
+        [1, 0, 1],
+        [1, 0, 1],
+    ])
+
+    tm2 = tm.annotate_cols(all_errors=hl.or_else(hl.agg.array_sum(implicated[tm.mendel_code]), [0, 0, 0]),
```

Comment:
since tm2 is not used again except the next line, can you make it `table3 = tm.annotate_cols().cols()`?

Reply:
yep, totally.